pkgs/development/haskell-modules/configuration-nix.nix

```diff
@@ -224,6 +224,9 @@ self: super: builtins.intersectAttrs super {
   # Test suite wants to connect to $DISPLAY.
   hsqml = dontCheck (addExtraLibraries (super.hsqml.override { qt5 = pkgs.qt5Full; }) [pkgs.libGLU pkgs.libGL]);
 
+  # Wants to check against a real DB, Needs freetds
+  odbc = dontCheck (addBuildDepends super.odbc [ pkgs.freetds ]);
```

Comment:
Try using `addExtraLibraries` instead of `addBuildDepends` which would avoid propagating.